frontend/src/components/pages/CampCreation/CampCreationFooter.tsx

```diff
@@ -6,18 +6,41 @@ export type CampCreationFooterProps = {
   currentStep: CampCreationPages;
   isCurrentStepCompleted: boolean;
   handleStepNavigation: (stepsToMove: number) => void;
+  isEditingCamp: boolean;
+  createUpdateCamp: (
+    isPublishedCamp: boolean,
+    isNewCamp: boolean,
+  ) => Promise<void>;
 };
 
 const CampCreationFooter = ({
   currentStep,
   isCurrentStepCompleted,
   handleStepNavigation,
+  createUpdateCamp,
+  isEditingCamp,
 }: CampCreationFooterProps): React.ReactElement => {
-  const onNextStep = () => {
+  const onNextStep = async () => {
     handleStepNavigation(1);
+
     if (currentStep === CampCreationPages.RegistrationFormPage) {
-      // eslint-disable-next-line no-console
-      console.log("Publish camp."); // Temporary.
+      // Publishing a new camp
+      if (!isEditingCamp) {
+        createUpdateCamp(true, true);
+      } else {
+        // Publishing a draft camp (should update the camp, not create a new camp)
+        createUpdateCamp(true, false);
+      }
```

Comment:
I would push for these arguments to be a type, it's pretty difficult to parse what's these mean when reading the code below

Alternatively, an easier fix would be something like:
```ts
const isPublished = true;
const isNewCamp = !isEditingCamp;

createUpdateCamp(isPublished, isNewCamp);
```